pkg/cloud/cloud.go

```diff
@@ -119,6 +122,10 @@ func (c *cloud) CreateFileSystem(ctx context.Context, volumeName string, fileSys
 		lustreConfiguration.SetExportPath(fileSystemOptions.S3ExportPath)
 	}
 
+	if fileSystemOptions.DeploymentType != "" {
+		lustreConfiguration.SetDeploymentType(fileSystemOptions.DeploymentType)
```

Comment:
should we validate the deploymenttype or let fsx API throw an error if the deploymenttype is invalid? The driver will retry forever , so it could prevent some useless api calls in case somebody makes a typo...

Reply:
In case of a typo in deployment type, the validation error is throw by FSx API model at the client side. It is redundant to do additional validation here. In my opinion, CSI driver code, as a wrapper around the FSx API,  should behave as a passthrough without reinventing any validation logic already implemented in FSx API model.